Implement simplified Schedule form

maurofmferrao opened the pull request:
relates to xibosignageltd/xibo-private#1017

maurofmferrao committed:
Implement simplified Schedule form
relates to xibosignageltd/xibo-private#1017

PeterMis reviewed (dismissed):
Looks good to me I think.

We probably want a migration for the 3 new settings with default values and expected userSee/userChange parameters, other than that it makes sense to me 👍

maurofmferrao committed:
Update Settings text and migration for new fields
 relates to xibosignageltd/xibo-private#1017

maurofmferrao committed:
Some fixes for steps behaviour
relates to xibosignageltd/xibo-private#1017

- Reset steps when changing Event type in add form
- Clear display groups for previous selected sync/normal events

maurofmferrao committed:
Tweaks and fixes
relates to xibosignageltd/xibo-private#1017

 - Fix Schedule now from Layout Editor
 - Exclude Sync Event for Schdule from Displays
 - Hide Event Type for Media/Playlist Schedule
 - Added User Features for Schedule
 - Added Schedule now to Display Group page
 - Remove migration for CMS schedule settings

PeterMis commented on lib/Controller/DisplayGroup.php:
```diff
@@ -399,6 +399,24 @@ public function grid(Request $request, Response $response)
+                $group->buttons[] = ['divider' => true];
+
+                $group->buttons[] = array(
+                    'id' => 'display_button_schedule',
```
It does not really matter, but should that be `displaygroup_button_schedule` ?
just to keep the same naming convention

PeterMis reviewed (commented):
Looks good, just minor comments below

PeterMis commented on lib/Controller/Settings.php:
```diff
@@ -643,6 +643,21 @@ public function update(Request $request, Response $response)
             $this->getConfig()->changeSetting('WHITELIST_LOAD_BALANCERS', $sanitizedParams->getString('WHITELIST_LOAD_BALANCERS'));
         }
 
+        if ($this->getConfig()->isSettingEditable('SCHEDULE_WITH_GEO')) {
```
Do we still need those?
since we moved them to user features, we can probably remove this from settings?

maurofmferrao committed:
Tweaks from code review
relates to xibosignageltd/xibo-private#1017

maurofmferrao committed:
Fix for start time for specific Daypart
relates to xibosignageltd/xibo-private#1017

maurofmferrao committed:
Schedule: End date validate and hide for command
relates to xibosignageltd/xibo-private#1017

dasgarner reviewed (approved):
Approved, but we need to confirm which scheduled features should be off by default

ifarzana commented:
TEST: /Schedule END

ifarzana commented:
TEST: login.cy.js END
